flask_oidc/__init__.py

```diff
@@ -486,268 +156,18 @@ def require_login(self, view_func):
         .. versionadded:: 1.0
            This was :func:`check` before.
         """
+
         @wraps(view_func)
         def decorated(*args, **kwargs):
-            if g.oidc_id_token is None:
-                return self.redirect_to_auth_server(request.url)
+            if session.get("token") is None:
+                redirect_uri = url_for(
+                    "_oidc_callback", _scheme="https", _external=True
+                )
+                return self.oauth.oidc.authorize_redirect(redirect_uri)
             return view_func(*args, **kwargs)
-        return decorated
-    # Backwards compatibility
-    check = require_login
-    """
-    .. deprecated:: 1.0
-       Use :func:`require_login` instead.
-    """
-
-    def require_keycloak_role(self, client, role):
-        """
-        Function to check for a KeyCloak client role in JWT access token.
-
-        This is intended to be replaced with a more generic 'require this value
-        in token or claims' system, at which point backwards compatibility will
-        be added.
-
-        .. versionadded:: 1.5.0
```

Comment:
This method is public and should probably be kept, as some apps may be using it (even though I don't think any Fedora apps use it, but flask-oidc is used outside Fedora too).